inst/test-expected_behavior_gamma_derivatives.R

```diff
@@ -0,0 +1,134 @@
+
+
+library(HistogramZoo)
+library(BoutrosLab.plotting.general)
+
+# Preamble ----------------------------------------------------------------
+# Testing gamma derivatives
+
+# Plotting function -------------------------------------------------------
+
+distribution_colours <- c(
+  "norm" = "darkorange",
+  "gamma" = "chartreuse4",
+  "gamma_flip" = "chartreuse3",
+  "unif" = "darkorchid4"
+)
+
+metric_lty <- c(
+  "mle" = 1,
+  "jaccard" = 2,
+  "intersection" = 3,
+  "chisq" = 4,
+  "ks" = 5,
+  "mse" = 6
+)
+
+plot_fit_distributions <- function(res, x, empirical_data){
+
+  plotting_data <- do.call(rbind.data.frame, lapply(res, function(res_i){
+    data.frame(
+      "x" = x,
+      "y" = res_i$dens(x, mpar = res_i$par, scale = F),
+      "metric" = res_i[['metric']],
+      "distribution" = res_i[['dist']],
+      "tag" = paste0(res_i[['metric']], "-", res_i[['dist']])
+    )
+  }))
+
+  plt <- create.scatterplot(
+    y ~ x,
+    plotting_data,
+    # plotting data
+    groups = plotting_data$tag,
+    type = "l",
+    col = distribution_colours[unique(plotting_data$distribution)],
+    lty = metric_lty[unique(plotting_data$metric)],
+    # Axes
+    xlab.cex = 1,
+    xlab.label = "x",
+    ylab.cex = 1,
+    ylab.label = "Data",
+    xaxis.cex = 1,
+    yaxis.cex = 1,
+    xlimits = c(min(x), max(x)),
+    # Adding points
+    add.points = T,
+    points.x = x,
+    points.y = empirical_data,
+    points.col = "red"
+  )
+
+  return(plt)
+}
+
+find_midpoint <- function(start, end){
+  (start + end)/2
+}
+
+# Gamma testing -----------------------------------------------------------
+
```

Comment:
Eventually this could get moved somewhere else.. possibly a vignette. We probably want to cleanup the non-essential stuff in `inst/` before `v2.0.0`.

Reply:
Yes, I couldn't think of a way to automate testing whether the behavior of truncated gamma/gamma_flip and distribution fitting was expected - hence this script that plotted every iteration of my refactoring.

I added an issue to clean up inst/ and possibly generate new vignettes - we can talk about this next meeting if you like :)

Do you mean `v3.0.0`?